### PR TITLE
Add material FAB

### PR DESCRIPTION
**Pull request by szakarias**

### CODE REVIEW

File: app/lib/frontend/dom/material.dart [hunk 1]
```diff
@@ -54,6 +54,44 @@ d.Node button({
   );
 }
 
+/// Renders a material floating action button (FAB) element.
+/// The FAB is 48x48px by default. If [fabMini] is `true` the FAB will be
+/// 40x40px.
+//
+// ignore: non_constant_identifier_names
+d.Node FAB({
```
Comment (jonasfj) [inline app/lib/frontend/dom/material.dart:62]:
```suggestion
d.Node floatingActionButton({
```

**Review by jonasfj — APPROVED** [review]

File: app/lib/frontend/dom/material.dart (hunk 1 above)
Comment (szakarias) [reply app/lib/frontend/dom/material.dart]: Done.

File: third_party/material/app.scss [hunk 2]
```diff
@@ -9,3 +9,5 @@
 @use '@material/ripple/mdc-ripple';
 @use '@material/select/mdc-select';
 @use '@material/textfield/mdc-text-field';
+@use "@material/fab";
```
Comment (isoos) [inline third_party/material/app.scss:12]: Let's keep the alphabetical order, and also the detailed pattern here (`@material/fab/mdc-fab`?).

File: third_party/material/app.scss [hunk 3]
```diff
@@ -9,3 +9,5 @@
...
+@include fab.core-styles;
```
Comment (isoos) [inline third_party/material/app.scss:13]: Is this needed? The reference file uses only fab:
https://github.com/material-components/material-components-web/blob/master/packages/material-components-web/material-components-web.scss#L35

File: third_party/material/initialize.sh [hunk 4]
```diff
@@ -14,6 +14,7 @@ EOF
 
 npm install "@material/auto-init@${VERSION}"
 npm install "@material/button@${VERSION}"
+npm install "@material/fab@${VERSION}"
```
Comment (isoos) [inline third_party/material/initialize.sh:17]: Don't we also need a JS initialization for this? Maybe not, as I couldn't find any traces here:
https://github.com/material-components/material-components-web/blob/master/packages/material-components-web/index.ts

**Review by isoos — CHANGES_REQUESTED** [review]

File: third_party/material/app.scss (hunk 2 above)
Comment (szakarias) [reply third_party/material/app.scss:12]: SG.

File: third_party/material/app.scss (hunk 3 above)
Comment (szakarias) [reply third_party/material/app.scss:13]: It seems not when `/mdc-fab` is added in the end of the `@use`. 

In https://material.io/components/buttons-floating-action-button/web#using-fabs it said 
```
@use "@material/fab";

@include fab.core-styles;
```

But ``` @material/fab/mdc-fab ``` seems to work so let's keep that. :-)

File: third_party/material/initialize.sh (hunk 4 above)
Comment (szakarias) [reply third_party/material/initialize.sh:17]: I think not. In my example it seems to work without initialization.

**Review by isoos — APPROVED** [review]

**Pull request merged**